internal/engine/executor.go

```diff
@@ -202,10 +202,10 @@ func (e *Executor) prepAndEvalEntityEvent(ctx context.Context, inf *entities.Ent
 	// TODO: clean this up once we get rid of provider name
 	var provider *db.Provider
 	var err error
-	if inf.ProviderID == uuid.Nil {
-		provider, err = e.providerStore.GetByName(ctx, inf.ProjectID, inf.Provider)
-	} else {
+	if inf.ProviderID != uuid.Nil {
```

Comment:
Swapped this condition because it is now expected that the Provider ID path will the more commonly used one